ash

```diff
@@ -1,624 +1,129 @@
 fi
 
-#
-# Now that all the jobs are complete, display a final report of
-# the return code status for each job that was run.
-#
-i=0
-echo -e "${CYAN}Jobs return code report:${NC}"
-for pid in "${JOBS[@]}"; do
-  REPORT_COLOR=${GREEN}
-  if [ ${JOBS_RC[${i}]} -ne 0 ]; then
-    REPORT_COLOR=${RED}
-    if [ ${JOBS_RC[${i}]} -gt ${HIGHEST_RC} ]; then
-      HIGHEST_RC=${JOBS_RC[${i}]}
-    fi
-  else
-    REPORT_COLOR=${GREEN}
-  fi
-  printf "${REPORT_COLOR}%32s${CYAN} : %3d${NC}\\n" "${JOB_NAMES[${i}]}" "${JOBS_RC[${i}]}"
-  i=$i+1
-done
-
-# Cleanup any previous file
-rm -f "${OUTPUT_DIR}"/"${AGGREGATED_RESULTS_REPORT_FILENAME}"
+# Create the output directory if it doesn't exist, otherwise the bind mount of the
+# OUTPUT_DIR will fail.
+if [ ! -d "${OUTPUT_DIR}" ]; then
+  mkdir -p "${OUTPUT_DIR}"
+fi
 
-# if an extension was not found, no report file will be in place, so skip the final report
-if [[ $(find "${OUTPUT_DIR}/work" -iname "*_report_result.txt" | wc -l | awk '{print $1}') -gt 0 ]];
-then
-  # Aggregate the results output files
-  for result in "${OUTPUT_DIR}"/work/*_report_result.txt;
-  do
-    echo "#############################################" >> "${OUTPUT_DIR}"/"${AGGREGATED_RESULTS_REPORT_FILENAME}"
-    echo "Start of  ${result}" >> "${OUTPUT_DIR}"/"${AGGREGATED_RESULTS_REPORT_FILENAME}"
-    echo "#############################################" >> "${OUTPUT_DIR}"/"${AGGREGATED_RESULTS_REPORT_FILENAME}"
-    cat "${result}" >> "${OUTPUT_DIR}"/"${AGGREGATED_RESULTS_REPORT_FILENAME}"
-    echo "#############################################" >> "${OUTPUT_DIR}"/"${AGGREGATED_RESULTS_REPORT_FILENAME}"
-    echo "End of  ${result}" >> "${OUTPUT_DIR}"/"${AGGREGATED_RESULTS_REPORT_FILENAME}"
-    echo -e "#############################################\n\n" >> "${OUTPUT_DIR}"/"${AGGREGATED_RESULTS_REPORT_FILENAME}"
-  done
+# Resolve the absolute paths
+SOURCE_DIR="$(cd "$SOURCE_DIR"; pwd)"
+OUTPUT_DIR="$(cd "$OUTPUT_DIR"; pwd)"
 
-  # Cleanup work directory containing all temp files
-  rm -rf "${OUTPUT_DIR}"/work
+# Resolve the OCI_RUNNER
+RESOLVED_OCI_RUNNER=${OCI_RUNNER:-$(command -v finch || command -v docker || command -v nerdctl || command -v podman)}
 
-  echo -e "${GREEN}\nYour final report can be found here:${NC} ${OUTPUT_DIR}/${AGGREGATED_RESULTS_REPORT_FILENAME}"
+# If we couldn't resolve an OCI_RUNNER, exit
+if [[ "${RESOLVED_OCI_RUNNER}" == "" ]]; then
+    echo "Unable to resolve an OCI_RUNNER -- exiting"
+    exit 1
+# else, build and run the image
 else
-  echo -e "${GREEN}No extensions were found, nothing to scan at the moment.${NC}"
-fi
-
-END_TIME=$(date +%s)
-TOTAL_EXECUTION=$((END_TIME-START_TIME))
-
-echo -e "${CYAN}ASH execution completed in ${TOTAL_EXECUTION} seconds.${NC}"
+    if [ "${DEBUG}" = "YES" ]; then
+      set -x
+    fi
+    echo "Resolved OCI_RUNNER to: ${RESOLVED_OCI_RUNNER}"
+
+    # Build the image if the --no-build flag is not set
+    if [ "${NO_BUILD}" = "NO" ]; then
```

Comment:
If you use the -q option then it takes some time between the message:
Resolved OCI_RUNNER to: xxxx
and the actual ASH execution.. so it seems that it hang, probably adding a message saying that build process is on the way or something like that. On other parts of ASH there are message telling you what it's actually being done

Reply:
Added